services/src/main/java/org/keycloak/email/DefaultEmailSenderProvider.java

```diff
@@ -107,8 +107,11 @@ public void send(Map<String, String> config, String address, String subject, Str
                 setupTruststore(props);
             }
 
-            props.setProperty("mail.smtp.timeout", "10000");
-            props.setProperty("mail.smtp.connectiontimeout", "10000");
+            String timeout = config.get("timeout");
+            String connectionTimeout = config.get("connectionTimeout");
+
+            props.setProperty("mail.smtp.timeout", timeout != null ? timeout : "10000");
+            props.setProperty("mail.smtp.connectiontimeout", connectionTimeout != null ? connectionTimeout : "10000");
```

Comment:
You can use `getOrDefault()` here to fall back to a default value, and likely just inline that for brevity.
```suggestion
            props.setProperty("mail.smtp.timeout", config.getOrDefault("timeout", "10000"));
            props.setProperty("mail.smtp.connectiontimeout", config.getOrDefault("connectionTimeout", "10000"));
```